### PR TITLE
Detailed expalin

### PR DESCRIPTION
**Pull request by haim-kermany**
1. exposing the detailed explain to the cli, as a flag
2. sunseting the debug output format

### CODE REVIEW

File: cmd/analyzer/expected_out/acl_testing3_detailed_explain.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Explaining connectivity from 10.240.10.4 to vsi2-ky within test-vpc1-ky
```
Comment (ShiriMoran) [inline cmd/analyzer/expected_out/acl_testing3_detailed_explain.txt:1]: I assume this is a renaming of an old test and it was verified that the results are identical to the new file?

File: cmd/analyzer/subcmds/explain.go [hunk 2]
```diff
@@ -53,6 +54,7 @@ func NewExplainCommand(args *inArgs) *cobra.Command {
 	cmd.Flags().Int64Var(&args.eSrcMaxPort, srcMaxPortFlag, netp.MaxPort, "maximum source port for connection description")
 	cmd.Flags().Int64Var(&args.eDstMinPort, dstMinPortFlag, netp.MinPort, "minimum destination port for connection description")
 	cmd.Flags().Int64Var(&args.eDstMaxPort, dstMaxPortFlag, netp.MaxPort, "maximum destination port for connection description")
+	cmd.Flags().BoolVar(&args.detailExplain, detailFlag, false, "adds a section with a list of all relevant allow/deny rules")
```
Comment (ShiriMoran) [inline cmd/analyzer/subcmds/explain.go:57]: Didn't we agree that the default will be true?
I think it should be true

File: pkg/ibmvpc/connectivityAnalysis_test.go [hunk 3]
```diff
@@ -1,443 +0,0 @@
-/*
```
Comment (ShiriMoran) [inline pkg/ibmvpc/connectivityAnalysis_test.go:1]: Why was this file removed?

File: pkg/vpcmodel/explainabilityPrint.go [hunk 4]
```diff
@@ -116,7 +115,7 @@ func explainMissingCrossVpcRouter(src, dst string, connQuery *connection.Set) st
 // 2 is printed only when the connection is blocked. It is redundant when the entire path ("3") is printed. When
 // the connection is blocked and only part of the path is printed then 2 is printed so that the relevant information
 // is provided regardless of where the is blocking
-// 4 is printed only in debug mode
+// 4 is printed only in detail mode
```
Comment (ShiriMoran) [inline pkg/vpcmodel/explainabilityPrint.go:118]:
```suggestion
// 4 is printed only in detailed mode
```

**Review by ShiriMoran — COMMENTED** [review]
Looks good. Few small comments

File: cmd/analyzer/expected_out/acl_testing3_detailed_explain.txt (hunk 1 above)
Comment (haim-kermany) [reply cmd/analyzer/expected_out/acl_testing3_detailed_explain.txt:1]: yes

File: cmd/analyzer/subcmds/explain.go (hunk 2 above)
Comment (haim-kermany) [reply cmd/analyzer/subcmds/explain.go:57]: ziv asked for false (after karen comment) we can discuss it again

File: pkg/ibmvpc/connectivityAnalysis_test.go (hunk 3 above)
Comment (haim-kermany) [reply pkg/ibmvpc/connectivityAnalysis_test.go:1]: Adi agreed it is not necessary.

File: pkg/vpcmodel/explainabilityPrint.go (hunk 4 above)
Comment (haim-kermany) [reply pkg/vpcmodel/explainabilityPrint.go]: fixed

File: cmd/analyzer/subcmds/explain.go (hunk 2 above)
Comment (ShiriMoran) [reply cmd/analyzer/subcmds/explain.go:57]: ok, lets discuss

**Comment by ShiriMoran** [comment]
LGTM - only the default issue remains

File: pkg/ibmvpc/analysis_output_test.go [hunk 5]
```diff
@@ -345,22 +348,6 @@ var tests = []*vpcGeneralTest{
...
-		inputConfig: "sg_testing1_new",
-		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints},
-		format:      vpcmodel.Debug,
-	},
-	{
-		inputConfig: "demo_with_instances",
-		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints},
-		format:      vpcmodel.Debug,
-	},
-	// disable drawio tests until supported with VPE
```
Comment (adisos) [inline pkg/ibmvpc/analysis_output_test.go:363]: please verify these tests exist at least with test output format

File: pkg/ibmvpc/connectivityAnalysis_test.go (hunk 3 above)
Comment (adisos) [reply pkg/ibmvpc/connectivityAnalysis_test.go:1]: Should we consider keeping the test, but changing the expected attributes to be compared with expected connectivity not as textual hard-coded output? 
can be in a separate issue/PR.

**Review by adisos — COMMENTED** [review]
looks good, just two small comments.

File: pkg/ibmvpc/analysis_output_test.go (hunk 5 above)
Comment (haim-kermany) [reply pkg/ibmvpc/analysis_output_test.go:363]: the same three tests are also in text format

**Review by zivnevo — DISMISSED** [review]

File: pkg/ibmvpc/connectivityAnalysis_test.go (hunk 3 above)
Comment (haim-kermany) [reply pkg/ibmvpc/connectivityAnalysis_test.go:1]: I dont see the point, much more complex connectivity cases are tested, and the output is checked by text.
the tests of this files are basically redundant, and more high maintained.

Comment (adisos) [reply pkg/ibmvpc/connectivityAnalysis_test.go:1]: these are more local tests, that do not run the entire flow.. it may be easier for debug and for capturing what is tested in this context, as opposed to e2e tests with json inputs.

**Review by adisos — APPROVED** [review]

File: pkg/ibmvpc/connectivityAnalysis_test.go (hunk 3 above)
Comment (haim-kermany) [reply pkg/ibmvpc/connectivityAnalysis_test.go]: I restored the file, and hard-codded changed the expected result.
Now we can later move the the expected result to files, and write code that update these file.

**Review by adisos — APPROVED** [review]
